src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -126,7 +126,7 @@ public AbstractJob createJob(final CitizenData citizen)
     @Override
     public boolean neededForWorker(@Nullable final ItemStack stack)
     {
-        return Utils.isMiningTool(stack);
+        return Utils.isMiningTool(stack) || neededResources.containsKey(stack.getUnlocalizedName());
```

Comment:
We should make a separate issue, where that is fixed. You don't want to pile on too much stuff to an already existing PR.